base/number.jl

```diff
@@ -314,9 +319,12 @@ zero(::Type{Union{}}, slurp...) = Union{}(0)
     one(T::type)
 
 Return a multiplicative identity for `x`: a value such that
-`one(x)*x == x*one(x) == x`.  Alternatively `one(T)` can
-take a type `T`, in which case `one` returns a multiplicative
-identity for any `x` of type `T`.
+`one(x)*x == x*one(x) == x`. If the multiplicative identity can
+be known based on the type alone, then a type may be given as
+an argument to `one` (e.g. `one(Int)` will work because the
+multiplicative identity is the same for all instances of `Int`,
+but `one(Matrix{Int})` is not defined because vectors of
+different lengths have different multiplicative identies.)
```

Comment:
```suggestion
but `one(Matrix{Int})` is not defined because matrices of
different shapes have different multiplicative identies.)
```

If you really want to go with `Matrix`